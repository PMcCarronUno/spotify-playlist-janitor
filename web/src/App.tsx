import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { RecoilRoot } from "recoil";
import Modal from "react-modal";
import styled from "styled-components";

import { HOME, CATALOGUE, LOGIN, PLAYLIST, REGISTER, ID } from "shared/constants";
import { AppHeader } from "shared/components/app-header";
import { Home } from "pages/home";
import { Playlist } from "pages/playlist";
import { Catalogue } from "pages/catalogue";
import { Login } from "pages/login";
import { Register } from "pages/register";
import { AuthProvider } from "shared/components/auth-provider";

function App() {
  Modal.setAppElement("#root");
  return (
    <RecoilRoot>
      <Container>
        <Router>
          <AppHeader />
          <Routes>
            <Route
              path={HOME}
              element={
                <AuthProvider shouldBeAuthorised={true} redirectPath={LOGIN}>
                  <Home />
                </AuthProvider>
              }
            />
            <Route
              path={`${PLAYLIST}${ID}`}
              element={
                <AuthProvider shouldBeAuthorised={true} redirectPath={LOGIN}>
                  <Playlist />
                </AuthProvider>
              }
            />
            <Route
              path={CATALOGUE}
              element={
                <AuthProvider shouldBeAuthorised={true} redirectPath={LOGIN}>
                  <Catalogue />
                </AuthProvider>
              }
            />
            <Route
              path={LOGIN}
              element={
                <AuthProvider shouldBeAuthorised={false} redirectPath={HOME}>
                  <Login />
                </AuthProvider>
              }
            />
            <Route
              path={REGISTER}
              element={
                <AuthProvider shouldBeAuthorised={false} redirectPath={HOME}>
                  <Register />
                </AuthProvider>
              }
            />
          </Routes>
        </Router>
      </Container>
    </RecoilRoot>
  );
}

const Container = styled.div`
  display: block;
`;

export default App;
