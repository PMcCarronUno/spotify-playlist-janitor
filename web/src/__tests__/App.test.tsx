import React from "react";
import { render } from "@testing-library/react";
import App from "../App";

jest.mock("react-modal");

describe("<App />", () => {

  beforeAll(() => {
    process.env.LOCAL_ENVIRONMENT = 'https://localhost:5001';
  });
  
  it("should render the App", async () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
