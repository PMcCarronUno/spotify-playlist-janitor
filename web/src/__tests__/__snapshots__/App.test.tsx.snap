// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render the App 1`] = `
<div
  class="sc-khjLgN hVyvMh"
>
  <div
    class="sc-jTQDJr dDKoCf"
  >
    <div
      class="header-item"
    >
      <div
        class="sc-guDMob jLqGJD"
        data-testid="home"
        id="home"
      >
        Spotify Playlist Janitor
      </div>
    </div>
    <div
      class="header-item sign-out"
    />
  </div>
  <div
    class="sc-dJGLgI gsJBuZ"
  >
    <div
      class="sc-ifysJV kBvnXl"
    >
      <div
        class="row"
      >
        <div
          class="sc-beyTiQ gCCLUC"
        >
          Login
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="sc-dmyDmy dcXriK"
        >
          Log in with email/password:
        </div>
      </div>
      <form
        autocomplete="off"
      >
        <div
          class="row"
        >
          <div
            class="sc-iBdoyZ bBxOEj"
          >
            <div
              class="input-container"
            >
              <input
                data-testid="email-input"
                id="email-input"
                name="email"
                placeholder="Enter email..."
              />
              <label
                class=""
                for="email-input"
              >
                Email
              </label>
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="sc-iBdoyZ bBxOEj"
          >
            <div
              class="input-container"
            >
              <input
                data-testid="password-input"
                id="password-input"
                name="password"
                placeholder="Enter password..."
                type="password"
              />
              <label
                class=""
                for="password-input"
              >
                Password
              </label>
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <button
            class="sc-fsYfxw sc-qZruQ icqseV bmcJcf"
            data-testid="login-button"
            id="login-button"
          >
            Log In
          </button>
        </div>
      </form>
      <div>
        <button
          class="sc-fsYfxw sc-jsEeA-d icqseV figgMk"
          data-testid="register-button"
          id="register-button"
        >
          Register
        </button>
      </div>
    </div>
  </div>
</div>
`;
